sdk/lusid/utilities/api_client_builder.py

```diff
@@ -37,68 +35,11 @@ def __check_required_fields(object_to_check, fields):
                 f"please ensure that you have provided them directly, via a secrets file or environment "
                 f"variables")
 
-    @staticmethod
-    def __generate_access_token(configuration, okta_response_handler):
-        """
-        This function generates an access token by making a call to Okta
-
-        :param ApiConfiguration configuration: The configuration to use
-        :param typing.callable okta_response_handler: An optional function to handle the Okta response
-
-        :return: RefreshingToken api_token: A refreshing API token
-        """
-        # Encode credentials that may contain special characters
-        encoded_password = pathname2url(configuration.password)
-        encoded_client_id = pathname2url(configuration.client_id)
-        encoded_client_secret = pathname2url(configuration.client_secret)
-
-        # Prepare our authentication request
-        token_request_body = f"grant_type=password&username={configuration.username}" \
-            f"&password={encoded_password}&scope=openid client groups offline_access" \
-            f"&client_id={encoded_client_id}&client_secret={encoded_client_secret}"
-
-        headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
-
-        # extra request args
-        kwargs = {"headers": headers}
-
-        if configuration.proxy_config is not None:
-            kwargs["proxies"] = configuration.proxy_config.format_proxy_schema()
-
-        # use certificate if supplied
-        if configuration.certificate_filename is not None:
-            kwargs["verify"] = configuration.certificate_filename
-
-        # make request to Okta to get an authentication token
-        okta_response = requests.post(configuration.token_url, data=token_request_body, **kwargs)
-
-        if okta_response_handler is not None:
-            okta_response_handler(okta_response)
-
-        # Ensure that we have a 200 response code
-        if okta_response.status_code != 200:
-            raise ValueError(okta_response.json())
-
-        # convert the json encoded response to be able to extract the token values
-        okta_json = okta_response.json()
-
-        # Retrieve our api token from the authentication response
-        api_token = RefreshingToken(token_url=configuration.token_url,
-                                    client_id=encoded_client_id,
-                                    client_secret=encoded_client_secret,
-                                    initial_access_token=okta_json["access_token"],
-                                    initial_token_expiry=okta_json["expires_in"],
-                                    refresh_token=okta_json["refresh_token"],
-                                    proxies=kwargs.get("proxies", None),
-                                    certificate_filename=kwargs.get("verify", None))
-
-        return api_token
-
```

Comment:
Moved this from the ApiClientBuilder into the RefreshingToken. This gives the RefreshingToken the ability to generate a new access token from username and password anytime.